core/src/language/python.rs

```diff
@@ -621,6 +458,318 @@ impl Python {
+                .map(|name| format!(
+                    "    {} = \"{}\"",
+                    name.to_case(Case::Snake).to_uppercase(),
+                    name
+                ))
+                .collect::<Vec<String>>()
+                .join("\n")
+        )?;
+        writeln!(w)?;
+
+        let mut variant_class_names = vec![];
+        let mut variant_constructors = vec![];
+        let mut contains_unit_variant = false;
+        // write each of the enum variant as a class:
+        for variant in &shared.variants {
+            let variant_class_name = make_struct_name(&variant.shared().id.original);
+            variant_class_names.push(variant_class_name.clone());
+
+            match variant {
+                RustEnumVariant::Unit(unit_variant) => {
+                    contains_unit_variant = true;
+
+                    self.add_import("typing".to_string(), "Literal".to_string());
+                    let variant_name = format!("{}{}", shared.id.renamed, unit_variant.id.renamed);
+                    variants.push((variant_name.clone(), vec![]));
+                    writeln!(w, "class {variant_class_name}(BaseModel):")?;
+                    writeln!(
+                        w,
+                        "    {content_key} = Literal[\"{}\"]",
+                        // escape any double quotes in the variant name
+                        unit_variant.id.renamed.replace("\"", "\\\"")
+                    )?;
+                    self.gen_unit_variant_ctor(
+                        &mut variant_constructors,
+                        unit_variant,
+                        shared,
+                        tag_key,
+                        content_key,
+                    );
+                    writeln!(w)?;
+                }
+                RustEnumVariant::Tuple {
+                    ty,
+                    shared: variant_shared,
+                } => {
+                    self.add_import("typing".to_string(), "Literal".to_string());
+                    self.gen_tuple_variant_ctor(
+                        &mut variant_constructors,
+                        variant_shared,
+                        shared,
+                        variant_class_name.clone(),
+                        tag_key,
+                        content_key,
+                    );
+                    match ty {
+                        RustType::Generic { parameters, .. } => {
+                            let mut generic_parameters: Vec<String> = parameters
+                                .iter()
+                                .flat_map(|p| {
+                                    collect_generics_for_variant(p, &shared.generic_types)
+                                })
+                                .collect();
+                            dedup(&mut generic_parameters);
+                            if generic_parameters.is_empty() {
+                                self.add_import("pydantic".to_string(), "BaseModel".to_string());
```

Comment:
I think base model should imported for any type we pass through so I think we can push that to a higher level function instead of here.